Fix Python 3.12 compatibility in wscript

Python 3.12 no longer accepts type='string' in argparse.
Change to type=str to fix compatibility.

--- a/wscript
+++ b/wscript
@@ -935,10 +935,13 @@ def options(opt):
     for ident, option in sorted(build_options.items()):
         if option['action'] == 'store':
             default = option['default']
+            opt_type = option.get('type')
+            # Python 3.12+ requires type=str instead of type='string'
+            if opt_type == 'string':
+                opt_type = str
             group.add_option('--{0}'.format(ident),
                              action = option['action'],
                              dest   = ident,
-                             type   = option['type'])
+                             type   = opt_type)
 
     group.add_option('--lua',
                      action = 'store',
